File: apk/app/src/main/java/com/example/momentra/ui/shell/group/shared/GroupViewerAccess.kt
package com.example.momentra.ui.shell.group.shared

import com.example.momentra.data.api.GroupParticipantDto
import java.util.Locale

/** Viewer (`OBSERVER` / legacy `VIEWER`) is read-only for Quick Add creates and expense entry. */
object GroupViewerAccess {
    fun isReadOnlyRole(roleCode: String?): Boolean {
        val upper = roleCode?.trim()?.uppercase(Locale.US).orEmpty()
        return upper == "OBSERVER" || upper == "VIEWER"
    }

    fun isViewer(participants: List<GroupParticipantDto>, currentUserId: String?): Boolean {
        if (currentUserId.isNullOrBlank()) return false
        val me = participants.firstOrNull { it.userId == currentUserId } ?: return false
        return isReadOnlyRole(me.roleCode)
    }
}

File: momentra/momentra/Shell/GroupActive/Shared/GroupViewerAccess.swift
import Foundation

/// Viewer (`OBSERVER` / legacy `VIEWER`) is read-only for Quick Add creates and expense entry.
enum GroupViewerAccess {
    static func isReadOnlyRole(_ roleCode: String?) -> Bool {
        let upper = (roleCode ?? "").trimmingCharacters(in: .whitespacesAndNewlines).uppercased()
        return upper == "OBSERVER" || upper == "VIEWER"
    }

    static func isViewer(
        participants: [APIClient.GroupParticipantPayload],
        currentUserId: String?
    ) -> Bool {
        guard let currentUserId, !currentUserId.isEmpty else { return false }
        guard let me = participants.first(where: { $0.userId == currentUserId }) else { return false }
        return isReadOnlyRole(me.roleCode)
    }
}
